src/api/auth-identities.tsx

```diff
@@ -0,0 +1,109 @@
+      .catch(reject);
+  });
+};
+
+export const fetchIdentity = (
+  id: string,
+  authMethod: string,
+): Promise<LxdIdentity> => {
+  return new Promise((resolve, reject) => {
+    fetch(`/1.0/auth/identities/${authMethod}/${id}`)
+      .then(handleResponse)
+      .then((data: LxdApiResponse<LxdIdentity>) => resolve(data.metadata))
+      .catch(reject);
+  });
+};
+
+export const updateGroupsForIdentity = (
+  identity: Partial<LxdIdentity>,
+  groups: string[],
+): Promise<void> => {
+  return new Promise((resolve, reject) => {
+    fetch(
+      `/1.0/auth/identities/${identity.authentication_method}/${identity.id}`,
+      {
+        method: "PUT",
+        body: JSON.stringify({
+          groups,
+        }),
+      },
+    )
+      .then(handleResponse)
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+export const updateGroupsForIdentities = (
+  identities: Partial<LxdIdentity>[],
+  groupsForIdentities: Record<string, string[]>,
+): Promise<void> => {
+  return new Promise((resolve, reject) => {
+    void Promise.allSettled(
+      identities.map((identity) =>
+        updateGroupsForIdentity(
+          identity,
+          groupsForIdentities[identity.id ?? ""] || [],
+        ),
+      ),
+    )
+      .then(handleSettledResult)
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+export const updateIdentitiesForGroup = (
+  identities: LxdIdentity[],
+  group: LxdGroup,
+): Promise<void> => {
+  const allExistingIdentityIds = getIdentityIdsForGroup(group);
+  return new Promise((resolve, reject) => {
+    const existingIdentitiesForGroup = new Set(allExistingIdentityIds);
+    const newIdentitiesForGroup = new Set(
+      identities?.map((identity) => identity.id),
+    );
+    // Determine if any identities were removed
+    const identitiesRemoved = allExistingIdentityIds?.filter(
+      (identity) => !newIdentitiesForGroup.has(identity),
+    );
+    // Determin if any identities were added to the group
+    const identitiesAdded = identities.filter(
+      (identity) => !existingIdentitiesForGroup.has(identity.id),
+    );
+
+    const removeIdentityPromises =
+      identitiesRemoved?.map(async (identity) => {
+        // Need to fetch identity details to find existing groups
+        const existingGroups = (await fetchIdentity(identity, "oidc")).groups;
+
+        return updateGroupsForIdentity(
+          { authentication_method: "oidc", id: identity },
+          existingGroups?.filter(
+            (existingGroup) => existingGroup !== group.name,
+          ) || [],
+        );
+      }) || [];
+    const addIdentityPromises = identitiesAdded.map((identity) =>
+      updateGroupsForIdentity(identity, [
+        ...(identity.groups || []),
+        group.name,
+      ]),
+    );
+
+    void Promise.allSettled([...removeIdentityPromises, ...addIdentityPromises])
+      .then(handleSettledResult)
+      .then(resolve)
+      .catch(reject);
+  });
+};
```

Comment:
```suggestion
```
unused?

Reply:
yeah that came from the spike, forgot to remove it.